src/vendor_app/app/src/main/kotlin/cz/quanti/android/vendor_app/repository/purchase/impl/PurchaseFacadeImpl.kt

```diff
@@ -0,0 +1,68 @@
+package cz.quanti.android.vendor_app.repository.purchase.impl
+
+import cz.quanti.android.vendor_app.repository.card.CardRepository
+import cz.quanti.android.vendor_app.repository.purchase.PurchaseFacade
+import cz.quanti.android.vendor_app.repository.purchase.PurchaseRepository
+import cz.quanti.android.vendor_app.repository.purchase.dto.Purchase
+import cz.quanti.android.vendor_app.utils.BlockedCardError
+import cz.quanti.android.vendor_app.utils.VendorAppException
+import cz.quanti.android.vendor_app.utils.isPositiveResponseHttpCode
+import io.reactivex.Completable
+import io.reactivex.Observable
+
+class PurchaseFacadeImpl(
+    private val purchaseRepo: PurchaseRepository,
+    private val cardRepo: CardRepository
+) : PurchaseFacade {
+
+    override fun savePurchase(purchase: Purchase): Completable {
+        if (purchase.smartcard != null) {
+            return cardRepo.getBlockedCard(purchase.smartcard!!).flatMapCompletable { itsBlocked ->
+                throw BlockedCardError("This card is tagged as blocked on the server")
+            }.andThen(purchaseRepo.savePurchase(purchase))
+        } else {
+            return purchaseRepo.savePurchase(purchase)
+        }
+    }
+
+    override fun syncWithServer(): Completable {
+        return sendPurchasesToServer()
```

Comment:
Tady bacha, jak to máš naskládaný. Nejdřív pošli vouchery, pak smaž vouchery z db, pak pošli nfc a pak smaž nfc. Takhle když by failnula synchronizace u těch druhejch, tak pak ty první (vouchery) pošleš znova, ne?